File: CompoundViews/app/src/main/java/com/bhavyakaria/compoundviews/MainActivity.java
package com.bhavyakaria.compoundviews;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.os.Handler;
import android.view.View;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity implements ClickListener{

    CustomButton button_one;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        button_one = findViewById(R.id.button_one);
        button_one.setListener(this);
    }

    @Override
    public void onUploadButtonClicked(View v, int id) {
        if (id == button_one.getId()) {
            button_one.showProgressBar();
            final Handler handler = new Handler();
            handler.postDelayed(new Runnable() {
                @Override
                public void run() {
                    // Do something after 5s = 5000ms
                    button_one.hideProgressBar();
                    button_one.showStatusText("Your pan card has been uploaded");
                }
            }, 5000);

        }
    }

    @Override
    public void onPointerCaptureChanged(boolean hasCapture) {

    }
}

File: Services/app/src/main/java/com/bhavyakaria/services/MainActivity.kt
package com.bhavyakaria.services

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import com.bhavyakaria.services.java.ServicesJavaActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btn_java.setOnClickListener {
            startActivity(Intent(applicationContext, ServicesJavaActivity::class.java))
        }

        btn_kotlin.setOnClickListener {
            //startActivity(Intent(applicationContext, ))
        }
    }
}
